classes/local/aws_helper.php

```diff
@@ -61,4 +65,36 @@ public static function get_proxy_string() {
         }
         return $proxy;
     }
+
+    /**
+     * Configure the provided AWS client to route traffic via the moodle proxy for any hosts not excluded.
+     *
+     * @param AwsClient $client
+     * @return AwsClient
+     */
+    public static function configure_client_proxy(AwsClient $client): AwsClient {
+        $client->getHandlerList()->appendBuild(self::get_callable_middleware(), 'proxy_bypass');
+        return $client;
+    }
+
+    /**
+     * Generate a middleware higher order function to wrap the handler and append proxy configuration based on target.
+     *
+     * @return callable Middleware high order callable.
+     */
+    protected static function get_callable_middleware(): callable {
+        return function (callable $fn) {
+            return function (CommandInterface $command, RequestInterface $request = null) use ($fn) {
+                if (!empty($request)) {
+                    $target = $request->getUri()->__toString();
```

Comment:
I would prefer a string cast here if required. Using __methods is generally a smell of sorts.